SportsHub/SportsHub.Api/Controllers/UserController.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using SportsHub.Domain.Models;
+using SportsHub.Domain.Models.Constants;
+using System.Security.Claims;
+
+namespace SportsHub.Api.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class UserController : ControllerBase
+    {
+        [HttpGet("Public")]
+        public IActionResult Public()
+        {
+            return Ok("Public property");
+        }
+
+        [HttpGet("Admins")]
+        [Authorize]
+        public IActionResult AdminsEndpoint()
+        {
+            var currentUser = GetCurrentUser();
+            return Ok($"Hi, {currentUser.Username}, you are an Admin");
+        }
+
+        private User GetCurrentUser()
+        {
+            var identity = HttpContext.User.Identity as ClaimsIdentity;
```

Comment:
I think it better to add some controller helper for doing this

Reply:
I've tried to move the method to AppService layer but I can't because HttpContext is derived from ControllerBase class. Where should I create the controller helper?

Reply:
implemented